### DISCUSSION TITLE
Running shell command in an async task?

### DISCUSSION THREAD

**Discussion by emiltin** [socketry/async#247] [Q&A] [answered]
What's the best way to run a shell command in an async task? I would like the task to read stdout (perhaps with something like Open3.popen2e) as the command writes to it, so I can complete the task or timeout as needed.

**Comment by ioquatix** [comment, marked as answer]
With Async 2+ it just works, and you don't need any custom code, it's all non-blocking, including `Process.wait` et al, reading and writing to the child process (via pipes), etc. Let me know if you run into any issues. Feel free to create one task to read and one to write, etc, depending on your requirements.

**Reply by emiltin** [reply]
great, thank you for the answer!
